### ISSUE TITLE
SnomedTermGraphLoader must not load `FULL` release without considering effectiveTeim

### ISSUE THREAD

**Issue by jaykola** [IBM/FHIR#2682]
**Describe the bug**
At the moment, `SnomedTermGraphLoader` has logic for example [`processConceptsFile()` method](https://github.com/IBM/FHIR/blob/6d0a060e03419afa5c5abca878c68ef4537cb28c/fhir-term-graph/src/main/java/com/ibm/fhir/term/graph/loader/impl/SnomedTermGraphLoader.java#L90), which totally ignores `effectiveTime` when importing a `FULL` release. This will result in a a graph being built that is unpredictable given SNOMED CT RF2 release format is `append only log style` convention.  Please refer to https://confluence.ihtsdotools.org/display/DOCSTART/13.+Release+Schedule+and+File+Formats

TLDR version - Full release must not be loaded this way. Use `Snapshot` release instead.

**Environment**
Which version of IBM FHIR Server?
All versions - currently 4.8-SNAPSHOT

**To Reproduce**
1. Follow the resource link above to RF2 documentation
2. Import a given version of a FULL release, then import the SNAPSHOT release
3. Compare graphs generated or simply compare `below` or `isa` and expansion for a concept like `22298006 | Myocardial infarction` and see the difference.

**Expected behavior**
FULL release must not be loaded without considering `effectiveTime` and computing the `snapshot` from it.
Alternatively, simply load the SNAPSHOT release instead

**Additional context**
This could also help with idempotent imports - #2178 to an extent and also addresses the comment in https://github.com/IBM/FHIR/issues/1980#issuecomment-810564913

**Comment by JohnTimm** [comment]
@jaykola The initial implementation implicitly considers effectiveTime when loading the FULL release.  Each set of concepts for a given id are already sorted in chronological order by effectiveTime in the RF2 file and we take the most recent concept and check to see if it is active and then include it only if it is active for the most recent effectiveTime. What is the expected behavior?

**Comment by jaykola** [comment]
Hmm. not sure what you mean by concepts for a given id… Do you mean all the set of entries for a given concept? I am referring to the `concept` file in this instance. 

> the general idea of sorting by effectiveTime and only considering the latest entry by `active` is the expected behaviour. 

I thought I had followed the logic in the code all the way through and hadn’t spotted the sorting logic. I haven’t come across any guidance that the entries in the RF2 file must be published sorted in chronological order. At least this isn’t in the RF2 spec as far as I am aware, but could something the publishing tool used for the US release does as an aside… ☺️ I always remember doing a `max(effectiveTime)` on the `table` anyways during import just cos I have had to deal with SNOMED CT editions from different countries…

**Comment by JohnTimm** [comment]
@jaykola Yes, I mean "all of the entries for a given concept id". Sorry, that was my bad phrasing. My observation is that entries (even from vastly different effective times) are grouped together in ascending (chronological order) but I think this must be a faulty assumption in general. I think we can solve this by mandating the use of the snapshot release and documenting as such.

**Comment by jaykola** [comment]
@JohnTimm yep SNAPSHOT as default makes sense. The use of FULL release depends on if/how users eventually want to use multiple releases of SNOMED CT. But then, we go into other complexity around transitive closure across versions and complicated terminology server territory. But for that you’ve provided a RemoteTerminologyService provider - which works perfectly for now.

My suggestion would be that FULL + DELTA way of incremental updates can be revisited at another time if that use case emerges.

**Comment by JohnTimm** [comment]
@jaykola do you know why some entries for a given appear grouped together and others may be located in completely different places in the file? This doesn't seem consistent with append-only approach you mentioned in your original post.

**Comment by jaykola** [comment]
From memory - the RF2 spec says the entries must be append only format - as in every time there is a change, the change has to be logged as a separate entry that is appended (not updating a previous entry). This would fulfil the `append only log` principle. My suspicion is that the publishing tools are grouping them together as it is perhaps easier during an export. 

The real issue is that sometimes an `Edition` needs to be created by merging the International (INT) release with the country specific release (e.g. US). So there could be more than one concept file to merge before import and determining the latest entry. If I am not wrong, the US (NLM) make life easier for downstream users by publishing this INT+US pieces as a single release with a single concept file.
